stutterhold/Source/PhaseVocodeur/PhaseVocodeur.cpp

```diff
@@ -48,7 +65,95 @@ PhaseVocodeur::PhaseVocodeur(int frame_size, int hop_size, int ola_size, int n_f
     init();
 }
 
-// ==============================================================================
+/*
+ ==============================================================================
+ Methods for initialization
+ ==============================================================================
+ */
+
+void PhaseVocodeur::init()
+{
+    /* Call all init subroutines */
+    init_ola();
+    init_window();
+    init_fft();
+}
+
+void PhaseVocodeur::init_ola()
+{
+    /* Initialize Overlap-Add (OLA) containers and rw positions */
+    
+    // assign number of frames based on overlap
+    num_ola_frames = ceil(static_cast<float>(ola_size) / static_cast<float>(hop_size));
+    
+    // allocate memory
+    ola_in = juce::AudioBuffer<float> (num_ola_frames, ola_size);
+    ola_out = juce::AudioBuffer<float> (num_ola_frames, ola_size);
+    
+    // clear
+    // @JulianArmandVanasse: Why? Cause residual junk maybe?

```

Comment:
I've noticed you clear everything beforehand. Is this a cpp thing that we don't have to worry about in Python-land?